src/coreclr/jit/importer.cpp

```diff
@@ -10324,6 +10324,28 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 }
                 else
                 {
+// TODO: enable for X86 as well, it currently doesn't support memset/memcpy helpers
+// Then, get rid of GT_STORE_DYN_BLK entirely.
+#ifndef TARGET_X86
+                    const bool     isVolatile = (indirFlags & GTF_IND_VOLATILE) != 0;
+                    const unsigned helper     = opcode == CEE_INITBLK ? CORINFO_HELP_MEMSET : CORINFO_HELP_MEMCPY;
+#ifdef TARGET_64BIT
+                    op3 = gtNewCastNode(TYP_I_IMPL, op3, /* fromUnsigned */ true, TYP_I_IMPL);
+#endif
+                    if (isVolatile)
+                    {
+                        // Wrap with memory barriers: full-barrier + call + load-barrier
+                        impSpillSideEffects(true, CHECK_SPILL_ALL DEBUGARG("spilling side-effects"));
```

Comment:
I think this needs to be called before we pop `op1`, `op2` and `op3` -- probably need to move the pops into each of the if/else.

Reply:
Ah, right

Reply:
Addressed